go-controller/pkg/ovn/policy.go

```diff
@@ -222,33 +223,69 @@ func deleteACLPortGroup(portGroupName, direction, priority, match, action string
 	return nil
 }
 
-func addToPortGroup(ovnNBClient goovn.Client, portGroupName string, portInfo *lpInfo) error {
-	cmd, err := ovnNBClient.PortGroupAddPort(portGroupName, portInfo.uuid)
-	if err == nil {
-		if err = ovnNBClient.Execute(cmd); err != nil {
-			return fmt.Errorf("execute error adding port %s to port group %s (%v)", portInfo.name, portGroupName, err)
+func addToPortGroup(ovnNBClient goovn.Client, portGroupName string, ports ...*lpInfo) error {
+	cmds := make([]*goovn.OvnCommand, 0, len(ports))
+	for _, portInfo := range ports {
+		cmd, err := ovnNBClient.PortGroupAddPort(portGroupName, portInfo.uuid)
+		if err != nil {
+			if err != goovn.ErrorExist {
+				return fmt.Errorf("error preparing adding port %v to port group %s (%v)", portInfo.name, portGroupName, err)
+			}
 		}
-	} else if err != goovn.ErrorExist {
-		// Ignore goovn.ErrorExist to implement port "--may-exist" behavior
-		// i.e., if the port is already there, it's okay.
-		return fmt.Errorf("error adding port %s to port group %s (%v)", portInfo.name, portGroupName, err)
+		cmds = append(cmds, cmd)
```

Comment:
Not familiar with goovn but it seems to wierd to append cmd if `err != nil`, without knowing I would expect a `continue`

Reply:
Oh, duh, good catch.